### PR TITLE
all: remove debug-field from vm config

### PR DESCRIPTION
**Pull request by holiman**
Alternative to #27002 

This PR removes the `Debug` field from vmconfig, making it so that if a tracer is set, `debug=true` is implied.

### CODE REVIEW

File: core/vm/interpreter.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ import (
 
 // Config are the configuration options for the Interpreter
 type Config struct {
-	Debug                   bool      // Enables debugging
+	Debug                   bool      // Opcode logger
```
Comment (s1na) [inline core/vm/interpreter.go:28]: Aha if we remove this downstream code will break

```suggestion
        // Opcode logger
        // Deprecated: this field is not used internally anymore and will be removed.
	Debug                   bool
```

**Review by s1na — APPROVED** [review]
Looks good to me!

**Review by rjl493456442 — APPROVED** [review]
lgtm

File: core/vm/interpreter.go (hunk 1 above)
Comment (holiman) [reply core/vm/interpreter.go:28]: :angry: meh ok

File: core/vm/interpreter.go [hunk 2]
```diff
@@ -25,7 +25,9 @@ import (
 
 // Config are the configuration options for the Interpreter
 type Config struct {
-	Debug                   bool      // Enables debugging
+        // Opcode logger
+        // Deprecated: this field is not used internally anymore and will be removed.
+	Debug                   bool
```
Comment (holiman) [inline core/vm/interpreter.go:30]:
```suggestion
```
Let's drop it?

Comment (s1na) [reply core/vm/interpreter.go]: 👍

**Pull request merged**